static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-header-eur-support.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+import {
+    getSupporterPaymentRegion as geolocationGetSupporterPaymentRegion,
+    getSync as geolocationGetSync,
+} from 'lib/geolocation';
+import { updateAcquisitionData } from 'common/modules/commercial/acquisitions-ophan';
+
+const componentType = 'ACQUISITIONS_HEADER';
+const campaignId = 'AcquisitionsEurSupport';
+const abTestName = 'AcquisitionsHeaderEurSupport';
+const EURsupportURL = 'https://support.theguardian.com/eu';
+
+type variantName = 'control' | 'support_contribute';
+
+const modifySupportTheGuardianLink = (variant: variantName): void => {
+    [...document.querySelectorAll('.js-change-become-member-link')].forEach(
+        supportTheGuardianLink => {
+            if (!(supportTheGuardianLink instanceof HTMLAnchorElement)) {
+                return;
+            }
+
+            let supportTheGuardianURL = new URL(supportTheGuardianLink.href);
+
+            if (variant === 'support_contribute') {
+                supportTheGuardianURL = new URL(EURsupportURL);
```

Comment:
I wonder whether it would be better to just pass through either `supportTheGuardianLink.href` or `EURsupportURL` to this function depending on the variant, then you won't have to assign this variable a value then conditionally overwrite it

Reply:
Both approaches are equally valid. I prefer to keep this logic inside the auxiliary function, it make the main method more clear and additionally in that way we can have the extra flow check.

Reply:
👍